### PR TITLE
Intro to Substrate MOOC - Module 1

### PR DESCRIPTION
**Pull request by CrackTheCode016**
The first section of the Substrate MOOC.

Tracking issue: https://github.com/w3f/w3f-education/issues/60

## todo

- [x] ~~Why Learn Substrate?~~
- [x] What is Substrate?
- [x] History Behind Substrate
- [x] Substrate’s Design Choices

### CODE REVIEW

**Comment by CrackTheCode016** [comment]
Current blockers:  Need more info on history of Substrate (something we can concretely link if possible), will also specify relationships between various components (Cumulus, Polkadot, role of Substrate in the context of this etc)

@DrW3RK I don't think "Why Learn Substrate" is really providing much value, what do you think? It can be included as a subsection somewhere else if anything.

**Comment by CrackTheCode016** [comment]
Another note: we should probably explain Substrate node (or "outer node", aka the concept of calls to native/gossip to other nodes), either here, or in Module 2.

**Comment by CrackTheCode016** [comment]
@ DrW3RK This section is blank at first glance.  This is intentional, as I realized it would be better to complete core content first then revisit what is needed from here.  Any additions and elaborations are very much welcome!

File: docs/Substrate/section1/substrate-design.md [hunk 1]
```diff
@@ -3,4 +3,28 @@ id: substrate-design
...
+
+## What is a generic code base?
+
+At a basic level, a generic code base represents a way to define very reusable code.  The Substrate libraries use this design pattern to optionally leave some of the definitions of primitives up to the developer.
+
+Blockchains often have standard 'primitives':  Polkadot, Bitcoin, or Ethereum all have concepts and primitives that are commonplace, i.e., transactions, blocks, storage, and hashing algorithms.  Substrate's generic nature allows for these blockchain primitives to be decided by the implementor.
+
+ We only really need to define these primitives one time for the chain.  In FRAME, this is done using a Rust trait called `Config`, which uses associated types to group all relevant primitives together in a way defined only once for the entire blockchain.
+
+## Are there any drawbacks?
```
Comment (DrW3RK) [inline docs/Substrate/section1/substrate-design.md:20]: rephrasing the title

```suggestion
## What are the challenges?
```

File: docs/Substrate/section1/substrate-design.md [hunk 2]
```diff
@@ -3,4 +3,28 @@ id: substrate-design
...
+
+There can be a degree of complexity for the developers maintaining the Substrate codebase, as there is a learning curve.  If you are a developer working closely with Substrate libraries, then intimate knowledge of how Rust typing works is crucial.
```
Comment (DrW3RK) [inline docs/Substrate/section1/substrate-design.md:22]: I may be missing the context here. Is this a challenge for the ones who develop Substrate? or the ones who are developing on it? May be add more explanation on the challenges involved.

```suggestion
There can be a degree of complexity for the developers maintaining the Substrate codebase, as there is a learning curve.  If you are a developer working closely with Substrate libraries, then intimate knowledge of how Rust typing works is crucial.
```

File: docs/Substrate/section1/substrate-history.md [hunk 3]
```diff
@@ -3,4 +3,8 @@ id: substrate-history
 title: History Behind Substrate
 sidebar_label: History Behind Substrate
 description: Learn the history of how Substrate came about and its relevance to Polkadot.
----
\ No newline at end of file
+---
+
+Substrate, at its origins, was created very near the conception of the Polkadot network.  The two are usually inseparable - where there is Substrate, there is Polkadot.  Parachains that connect to Polkadot are also built using Substrate, generally using some abstractions such as FRAME and Cumulus.
```
Comment (DrW3RK) [inline docs/Substrate/section1/substrate-history.md:8]: There are Substrate solo chains that are successfully deployed with no connection to Polkadot. Updating it to reflect that Substrate development is sort of tightly coupled with Polkadot's.

```suggestion
Substrate, at its origins, was created very close to the conception of the Polkadot network.  The two are usually tightly coupled - where the improvements to Substrate funnel into Polkadot and vice versa.  Parachains that connect to Polkadot are also built using Substrate, generally using some abstractions such as FRAME and Cumulus.
```

File: docs/Substrate/section1/what-is-substrate.md [hunk 4]
```diff
@@ -1,10 +1,59 @@
 ---
 id: what-is-substrate
 title: Substrate - the backend of web3
```
Comment (DrW3RK) [inline docs/Substrate/section1/what-is-substrate.md:3]:
```suggestion
title: Substrate - powering Web3
```

File: docs/Substrate/section1/what-is-substrate.md [hunk 5]
```diff
@@ -1,10 +1,59 @@
...
 
-In order to get the most out of Substrate, you should have a good knowledge of computer science and basic blockchain concepts. Terminology like header, block, client, hash, transaction and signature should be familiar. Substrate is built on the [Rust Programming language](https://www.rust-lang.org/), which makes use of novel design patterns to enable development of code that is safe and fast. Although you don't need to know Rust to get started with Substrate, a good understanding of Rust will allow you to become a better Substrate developer.
\ No newline at end of file
+Substrate may be utilized to create a solo chain and parachain for a relay chain like Polkadot (respectively).  At its core, it is a set of Rust crates that provide the most basic primitives and protocols to implement a distributed state machine, such as a blockchain.
```
Comment (DrW3RK) [inline docs/Substrate/section1/what-is-substrate.md:10]:
```suggestion
Substrate may be utilized to create a solo chain or a parachain for a relay chain like Polkadot or Kusama.  At its core, it is a set of Rust crates that provide the most basic primitives and protocols to implement a distributed state machine, such as a blockchain.
```

File: docs/Substrate/section1/what-is-substrate.md [hunk 6]
```diff
@@ -1,10 +1,59 @@
...
+
+Another way to think about Substrate is similar to how a web developer may use/think about React.  Where React provides core libraries for manipulating the elements of the DOM via developer-defined components, Substrate also provides the base libraries and primitives needed to create a blockchain.  It was designed to not be opinionated via a highly generic codebase, enabling high customizability.
+
+:::
+
+## Substrate's Libraries
+
+Substrate, as mentioned, is a collection of Rust crates that define a generic way to implement a blockchain.  Although you will only use a few of these libraries directly in this course, know that everything you use is built upon them. 
+
+You will encounter two primary prefixes when developing with Substrate crates: `sc-` and `sp-`.  Crates prefixed with `sc-` usually refer to network-related functionalities, such as peer-to-peer networking or consensus.  The `sp-` prefix usually indicates that the library in question is for allowing the outer node and runtime to communicate.
```
Comment (DrW3RK) [inline docs/Substrate/section1/what-is-substrate.md:22]: sc is substrate client and sp is substrate primitive

**Review by DrW3RK — APPROVED** [review]
Minor fixes. Looks good to me!

File: docs/Substrate/section1/what-is-substrate.md (hunk 6 above)
Comment (CrackTheCode016) [reply docs/Substrate/section1/what-is-substrate.md]: Addressed 👍

File: docs/Substrate/section1/substrate-design.md (hunk 2 above)
Comment (CrackTheCode016) [reply docs/Substrate/section1/substrate-design.md]: Addressed 👍

**Comment by CrackTheCode016** [comment]
@DrW3RK Feedback addressed and conflicts fixed, feel free to merge or suggest anything else!

**Pull request merged**